articles/active-directory/develop/quickstart-v2-java-webapp.md

```diff
@@ -22,78 +22,90 @@ ms.custom: aaddev
-> 1. Select the **Authentication** menu, and then add the following information:
->    - In **Redirect URIs**, add `http://localhost:8080/msal4jsamples/secure/aad` and `https://localhost:8080/msal4jsamples/graph/users`.
->    - Select **Save**.
-> 1. On the left hand menu, choose **Certificates & secrets** and click on **New client secret** in the **Client Secrets** section:
->     
->    - Type a key description (of instance app secret).
->    - Select a key duration of **In 1 year**.
->    - When you click on **Add**, the key value will be displayed. 
->    - Copy the value of the key, you will need it later.
+
+## Register and download your quickstart app
+
+You have two options to start your quickstart application: express (Option 1), or manual (Option 2)
+
+### Option 1: Register and auto configure your app and then download your code sample
+
+1. Go to the [Azure portal - App registrations](https://portal.azure.com/#blade/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/RegisteredApps).
+1. Enter a name for your application and select **Register**.
+1. Follow the instructions to download and automatically configure your new application.
+If you want to use this automation, read the instructions in [App Creation Scripts](./AppCreationScripts/AppCreationScripts.md). Please note that the configuration of your code (Step 3) still needs to be done manually.
+
+### Option 2: Register and manually configure your application and code sample
+
+#### Step 1: Register your application
+
+To register your application and manually add the app's registration information to your solution, follow these steps:
+
+1. Sign in to the [Azure portal](https://portal.azure.com) using either a work or school account, or a personal Microsoft account.
+1. If your account gives you access to more than one tenant, select your account in the top right corner, and set your portal session to the desired Azure AD tenant.
+
+1. Navigate to the Microsoft identity platform for developers [App registrations](https://go.microsoft.com/fwlink/linkid=2083908) page.
+1. Select **New registration**.
+1. When the **Register an application** page appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `java-webapp`.
+   - Leave **Redirect URI** blank for now, and select **Register**.
+1. On the **Overview** page, find the **Application (client) ID** and the **Directory (tenant) ID** values of the application. Copy these values for later.
+1. Select the **Authentication** from the menu, and then add the following information:
+   - In **Redirect URIs**, add `http://localhost:8080/msal4jsamples/secure/aad` and `https://localhost:8080/msal4jsamples/graph/me`.
+   - Select **Save**.
+1. Select the **Certificates & secrets** from the menu and in the **Client secrets** section, click on **New client secret**:
+
+   - Type a key description (for instance app secret).
```

Comment:
Are you saying that 'app secret' is a description they could enter? It's a little confusing. I'd drop the entire parenthetical note.